### ISSUE TITLE
Error TS5052 without having "declarationDir" specified

### ISSUE THREAD

**Issue by Prior99** [s-panferov/awesome-typescript-loader#391]
With version the latest version (3.1.1) I am getting the error

```
ERROR in [at-loader] TS5052: Option 'declarationDir' cannot be specified without specifying option 'declaration'.
```

without having "declarationDir" specified in my `tsconfig.json`.

My `tsconfig.json` looks like this:

```json
{
    "compilerOptions": {
        "outDir": "./dist/",
        "sourceMap": true,
        "noImplicitAny": true,
        "module": "commonjs",
        "target": "es5",
        "jsx": "react",
        "pretty": true,
        "noUnusedParameters": true,
        "experimentalDecorators": true,
        "noUnusedLocals": true,
        "lib": ["dom", "es2017"]
    },
    "include": [
        "src/**/*.ts?",
        "config.ts",
        "typings/*"
    ]
}
```

and the relevant part of my webpack config (webpack 2.0) as follows:

```js
module.exports = {
    entry: "./src/index.tsx",
    output: {
        filename: "bundle.js",
        path: __dirname + "/dist",
        publicPath: "/dist"
    },
    devtool: "source-map",
    resolve: {
        extensions: [".js", ".jsx", ".ts", ".tsx", ".json"]
    },
    devServer: {
        // [...]
    },
    plugins: [
        // [...]
    ],
    module: {
        rules: [
            // [...]
            {
                test: /\.(tsx?|json)$/,
                exclude: /node_modules/,
                use: [
                    {
                        loader: "awesome-typescript-loader"
                    }
                ]
            },
            {
                test: /\.js$/,
                exclude: /node_modules/,
                use: [
                    {
                        loader: "source-map-loader"
                    }
                ]
            }
        ]
    }
};
```

With version 3.0.8 everything is working fine.

I am running Typescript version 2.2.0.

**Comment by jrmcdona** [comment]
Same here.  For some reason, it wasn't there yesterday and today it is failing.

**Comment by Prior99** [comment]
You can work around this issue by adding
```
"awesome-typescript-loader": "=3.0.8",
```

to your package.json which will set the version fixed to `3.0.8` which was working for me.

**Comment by giniedp** [comment]
has to do with the fix of https://github.com/s-panferov/awesome-typescript-loader/issues/247
simply set the `declaration` option when you specify the `declarationDir` option.

**Comment by Prior99** [comment]
I did not specify `declarationDir` at all. Anywhere. (Which is why I am wondering what causes this issue). Setting `declaration: true` in my tsconfig.json anyway will cause other problems when building without at-loader.

**Comment by FastAndFluid** [comment]
Seeing the same thing which wasn't there in 3.1.0 but the same error I reported in 3.1.0 which I thought 3.1.1 was released to fix is still there in  3.1.1:

 Child process failed to process the request:  TypeError: Cannot read property '0' of undefined
    at D:\myProject\node_modules\awesome-typescript-loader\src\checker\runtime.ts:413:29
    at Array.map (native)
    at processDiagnostics (D:\myProject\node_modules\awesome-typescript-loader\src\checker\runtime.ts:409:14)
    at D:\myProject\node_modules\awesome-typescript-loader\src\checker\runtime.ts:475:21
    at Object.send (D:\myProject\node_modules\awesome-typescript-loader\src\checker\runtime.ts:27:21)
    at Checker.req (D:\myProject\node_modules\awesome-typescript-loader\src\checker\checker.ts:101:21)
    at Checker.getDiagnostics (D:\myProject\node_modules\awesome-typescript-loader\src\checker\checker.ts:136:21)
    at Compiler.<anonymous> (D:\myProject\node_modules\awesome-typescript-loader\src\instance.ts:403:32)
    at next (D:\myProject\node_modules\tapable\lib\Tapable.js:140:14)
    at Compiler.<anonymous> (D:\myProject\node_modules\webpack\lib\CachePlugin.js:61:4)

ERROR in [at-loader] TS5052: Option 'declarationDir' cannot be specified without specifying option 'declaration'.

**Comment by Prior99** [comment]
@FastAndFluid I am experiencing the same issue with "Cannot read property '0' of undefined" on 3.1.0.

**Comment by FrozenPandaz** [comment]
I actually specify declaration: false and its giving me this error

**Comment by icfantv** [comment]
@FrozenPandaz `false` is the default value for `declaration`.

**Comment by s-panferov** [comment]
Should be fixed by awesome-typescript-loader@3.1.2
